### PR TITLE
Hash algorithm agility

### PR DESCRIPTION
**Pull request by tireddy2**

### CODE REVIEW

**Comment by bemasc** [comment]
Could you expand the rationale for this change?  The [Zone Digest registry](https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#zonemd-hash-algorithms) already provides hash agility here.

**Comment by tireddy2** [comment]
> Could you expand the rationale for this change? The [Zone Digest registry](https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#zonemd-hash-algorithms) already provides hash agility here.

Yes, I missed the zone digest registry; reverted the changes and added text to discuss hash algorithm agility in the security considerations section and the mandatory-to-implement algorithms for implementations.

File: draft-ietf-add-split-horizon-authority.xml [hunk 1]
```diff
@@ -311,8 +314,7 @@
             <li>"subdomains": An array containing the claimed subdomains, as
                 dot-separated names with the parent suffix already removed, in
                 canonical order.</li>
-            <li>"algorithm": The hash algorithm, identified by its IANA-registered
-                Mnemonic.</li>
+            <li>"algorithm": The hash algorithm, identified by the the ZONEMD Hash Algorithm.</li>
```
Comment (bemasc) [inline draft-ietf-add-split-horizon-authority.xml]: In the ZONEMD registry, there is no column labeled "Hash Algorithm".  Instead, there are columns labeled "Value" (number), "Description" (string), and "Mnemonic" (string), any of which could be encoded in JSON.  I think the description here probably needs to be precise about which column we are using.

File: draft-ietf-add-split-horizon-authority.xml [hunk 2]
```diff
@@ -221,7 +221,10 @@
         <li>A set of subdomains of this parent zone that are claimed by
             the named local resolver (potentially including the entire parent
             zone).</li>
-        <li>A ZONEMD Hash Algorithm (<relref section="5.3" target="RFC8976"/>).</li>
+        <li>A ZONEMD Hash Algorithm (<relref section="5.3" target="RFC8976"/>). 
+           For interoperability purposes implementations have to support the
+           mandatory to implement hash algorithms defined in
```
Comment (bemasc) [inline draft-ietf-add-split-horizon-authority.xml]:
```suggestion
           For interoperability purposes implementations MUST support the
           "mandatory to implement" hash algorithms defined in 
```

File: draft-ietf-add-split-horizon-authority.xml (hunk 1 above)
Comment (tireddy2) [reply draft-ietf-add-split-horizon-authority.xml]: Agreed, updated text as follows:
   *  "algorithm": The hash algorithm is represented as an unsigned integer from the ZONEMD Hash Algorithms registry ([RFC8976], Section 5.2).

**Pull request merged**